c_src/sqlite3_nif.c

```diff
@@ -632,6 +632,74 @@ exqlite_transaction_status(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
       autocommit == 0 ? make_atom(env, "transaction") : make_atom(env, "idle"));
 }
 
+static ERL_NIF_TERM
+exqlite_serialize(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+{
+    assert(env);
+
+    connection_t* conn = NULL;
+    unsigned char* buffer = NULL;
+    sqlite3_int64 buffer_size = 0;
+    ERL_NIF_TERM serialized;
+
+    if (argc != 1) {
+        return enif_make_badarg(env);
+    }
+
+    if (!enif_get_resource(env, argv[0], connection_type, (void**)&conn)) {
+        return make_error_tuple(env, "invalid_connection");
+    }
+
+    buffer = sqlite3_serialize(conn->db, "main", &buffer_size, 0);
```

Comment:
I've hard-coded the database name `"main"` here as that's enough for my intended usage. 
It could alternatively be taken as a parameter, and defaulted to `"main"` in `Exqlite.Sqlite3.serialize`.

Reply:
Let's allow this to be specified through args in `Sqlite3` and change the signature for `Sqlite3NIF` to accept a database name.